numpy/_core/src/umath/_scaled_float_dtype.c

```diff
@@ -861,6 +862,205 @@ sfloat_init_ufuncs(void) {
 }
 
 
+NPY_NO_EXPORT int
+sfloat_stable_sort_loop(
+        PyArrayMethod_Context *context,
+        char *const *data,
+        const npy_intp *dimensions,
+        const npy_intp *strides,
+        NpyAuxData *NPY_UNUSED(auxdata))
+{
+    npy_intp N = dimensions[0];
```

Comment:
Since this is partially to test our implementation, how about
```
assert(data[0] == data[1]);
assert(dimensions[0] == dimensions[1]);
assert(strides[0] == strides[1]);
assert(context->parameters->flags == NPY_SORT_STABLE);
```
(probably overkill... obviously can also just `return -1` in any of those cases).